src/node_report.cc

```diff
@@ -917,8 +920,17 @@ std::string TriggerNodeReport(Isolate* isolate,
     compact = per_process::cli_options->report_compact;
   }
 
-  report::WriteNodeReport(
-      isolate, env, message, trigger, filename, *outstream, error, compact);
+  bool network_disabled = env->options()->report_network_disabled;
```

Comment:
`env` could be nullptr here and in other places, so that can cause the segfault in the cctests. In that case you can fall back to `per_process::cli_options->per_isolate->per_env`. I think it should be fine not to acquire the `cli_options_mutex` in that case (e.g. when the report is triggered by a signal when Node.js is not running JavaScript, or when the Node.js instance is not yet fully bootstrapped/being torn down), since I don't recall any other code that allows other threads to mutate `report_network_disabled` or `per_env` in that state but it may also be safe to acquire a lock only in that case.

Reply:
That worked! I looked into it and it seems like I can completely rely on this rather than doing some kind of `if (env == nullptr)` check. Tests pass locally.

Reply:
in hindsight, this change broke things. Seems like I'm not able to set the option the same way.